website/docs/r/group.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: "azuread"
+page_title: "Azure Active Directory: azuread_group"
+sidebar_current: "docs-azuread-resource-azuread-group"
+description: |-
+  Manages a Group within Azure Active Directory.
+
+---
+
+# azuread_group
+
+Manages a Group within Azure Active Directory.
+
+-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to `Read and write all groups` within the `Windows Azure Active Directory` API.
+
+-> **NOTE:** Additionally, due to a limitation within the API, you have to assign **one** of the following Azure Active Directory Roles to the Service Principal to be able to delete Groups:
+
+* User Account Administrator
+* Company Administrator
+
+You can assign one of the required Azure Active Directory Roles with PowerShell. Please refer to [this documentation](https://docs.microsoft.com/en-us/powershell/module/azuread/add-azureaddirectoryrolemember) for more details.
```

Comment:
I think it'd be better to be platform agnostic and reference the PowerShell method of doing this here: https://docs.microsoft.com/en-us/cli/azure/role/assignment?view=azure-cli-latest#az-role-assignment-create - WDYT?

Reply:
Please note that you need to assign Azure AD directory roles to the Service Principal, that is not the same as an Azure Role Assignment. This assignment is currently only possible via the PowerShell Module "AzureAD" (available on Windows PowerShell or in the Azure Cloud Shell) or directly through the Graph API.

I'm going to add some more details to the docs to make this more clear.